internal/services/report_processor.go

```diff
@@ -113,6 +114,14 @@ func ProcessReport(msg *kafka.Message) {
 				k8s_object_name,
 			)
 
+			if workload_metrics, err := model.GetWorkloadMetricsForTimestamp(experiment_name, interval_end); err != nil {
+				log.Errorf("Error while checking for workload_metrics record: %s", err)
+				continue
+			} else if !reflect.ValueOf(workload_metrics).IsZero() {
```

Comment:
```suggestion
			} else if workload_metrics == (WorkloadMetrics{}) {
```
In case this works, we can avoid depending on `reflect`

Reply:
I guess I tried that but it did not worked.